Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -395,6 +393,28 @@ void AODProducerWorkflowDPL::fillMCParticlesTable(o2::steer::MCKinematicsReader&
       if (item != toStore.end()) {
         daughterL = item->second;
       }
+      float pX = (float)mcParticles[particle].Px();
+      float pY = (float)mcParticles[particle].Py();
+      float pZ = (float)mcParticles[particle].Pz();
+      float energy = (float)mcParticles[particle].GetEnergy();
+      // HACK to avoid FPE in expression columns. Affect only particles in the Beam Pipe.
+      // TO BE REMOVED asap
+      {
+        const float limit = 1e-4;
+        const float mom = TMath::Sqrt(pX * pX + pY * pY + pZ * pZ);
+        const float eta = 0.5f * TMath::Log((mom + pZ) / (mom - pZ));
+        if (TMath::Abs(eta) > 0.9) {
+          if (TMath::Abs((mom - pZ) / pZ) <= limit) {
+            pX = truncateFloatFraction(TMath::Sqrt((1.f + limit) * (1.f + limit) - 1.f) * pZ * 0.5, mMcParticleMom);
+            pY = truncateFloatFraction(TMath::Sqrt((1.f + limit) * (1.f + limit) - 1.f) * pZ * 0.5, mMcParticleMom);
+          }
+          if (TMath::Abs(energy - pZ) < limit) {
+            energy = truncateFloatFraction(pZ + limit, mMcParticleMom);
+          }
```

Comment:
could you elaborate what produces FPE? In any case, if you want to approximate the 
````
pT^2 = ( (1+limit)^2 - 1) pZ^2
````
then one should have 
````
pX = truncateFloatFraction(TMath::Sqrt((1.f + limit) * (1.f + limit) - 1.f) * pZ  / sqrt(2), mMcParticleMom);
...
````

Reply:
Here the FPE is observed when computing rapidity/pseudorapidity for particles that are in the beampipe. This hack is needed to bypass a limitation that will be solved in the future when we will be able to handle conditions in Gandiva for the computation of the expression columns.

Reply:
OK, but if you want to make `(mom - pZ) / pZ) = limit`, this means `pT^2 = ( (1+limit)^2 - 1) pZ^2` and with `pX=pY=pT/sqrt(2)` this gives `pX=pY=truncateFloatFraction(TMath::Sqrt((1.f + limit) * (1.f + limit) - 1.f) * pZ  / sqrt(2), mMcParticleMom);` (note the factor 1/sqrt(2) instead of 0.5)

Reply:
that's true, thanks for spotting this!

Reply:
nonetheless, the difference should be small i.e. `*0.7` vs `*0.5`